Client/Frontend/Browser/BrowserViewController/BVC+Rewards.swift

```diff
@@ -161,6 +161,43 @@ extension BrowserViewController {
         Preferences.NewTabPage.brandedImageShowed.value = false
         Preferences.NewTabPage.atleastOneNTPNotificationWasShowed.value = false
     }
+    
+    // MARK: - SKUS
+    
+    func paymentRequested(_ request: PaymentRequest, _ completionHandler: @escaping (_ response: PaymentRequestResponse) -> Void) {
+        if UIDevice.current.userInterfaceIdiom != .pad && UIApplication.shared.statusBarOrientation.isLandscape {
+            let value = UIInterfaceOrientation.portrait.rawValue
+            UIDevice.current.setValue(value, forKey: "orientation")
+        }
+        
+        if !rewards.ledger.isEnabled {
+            let enableRewards = SKUEnableRewardsViewController(
+                rewards: rewards,
+                termsURLTapped: { [weak self] in
+                    if let url = URL(string: DisclaimerLinks.termsOfUseURL) {
+                        self?.loadNewTabWithURL(url)
+                    }
+                }
+            )
+            present(enableRewards, animated: true)
+            return
```

Comment:
We need to `reject` the response here, can't let the promise unresolved if the `enableRewards` screen is dismissed.